tests/ludwig/modules/test_metric_modules.py

```diff
@@ -36,12 +37,17 @@ def test_rmspe_metric(preds: torch.Tensor, target: torch.Tensor, output: torch.T
     [
         (torch.arange(3), torch.arange(3, 6), 1, torch.tensor(-12.5)),
         (torch.arange(6).reshape(3, 2), torch.arange(6, 12).reshape(3, 2), 2, torch.tensor(-12.5)),
+        (torch.tensor([0.8]), torch.arange(1), 1, np.nan),
     ],
 )
 def test_r2_score(preds: torch.Tensor, target: torch.Tensor, num_outputs: int, output: torch.Tensor):
     metric = metric_modules.R2Score(num_outputs=num_outputs)
     metric.update(preds, target)
-    assert metric.compute() == output
+    computed_score = metric.compute()
+    if np.isnan(computed_score):
```

Comment:
Avoid logic in tests:

https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-logic-in-tests